src/main/scala/io/delta/execution/DeltaTableOperations.scala

```diff
@@ -170,6 +170,23 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     }
   }
 
+  protected def executeVacuum(
+      deltaLog: DeltaLog,
+      dryRun: Boolean,
+      retentionHours: Option[Double]): DataFrame = {
+    val sparkSession = self.toDF.sparkSession
+    val result = VacuumCommand.gc(sparkSession, deltaLog, dryRun, retentionHours)
+    if (dryRun) {
+      result
+    } else {
+      sparkSession.emptyDataFrame
+    }
+  }
   override protected lazy val sparkSession: SparkSession = self.toDF.sparkSession
+
+  protected lazy val deltaLog = (EliminateSubqueryAliases(self.toDF.queryExecution.analyzed) match {
```

Comment:
As a follow up, we should go from a deltaLog -> df, and not the other way around. This is fine for now